File: code/generate_strategies/RA(name)/java/java/function_collections_counter.java
// 功能类路径：src/main/java/com/example/function_collections_counter.java
package com.example;

import com.google.common.collect.LinkedHashMultiset;
import com.google.common.collect.Multiset;
import com.google.common.base.Splitter;
import java.util.List;
import java.util.ArrayList;

public class CounterCalculator {
    public String count_elements(String data) {
        if (data == null) return "failed";

        // 使用Guava的Splitter处理字符串分割并过滤空元素
        List<String> words = Splitter.onPattern("\\s+")
                .omitEmptyStrings()
                .splitToList(data);

        if (words.isEmpty()) return "failed";

        // 使用LinkedHashMultiset保持元素插入顺序
        Multiset<String> counter = LinkedHashMultiset.create();
        counter.addAll(words);

        // 构建结果字符串
        List<String> entries = new ArrayList<>();
        for (Multiset.Entry<String> entry : counter.entrySet()) {
            entries.add(entry.getElement() + ":" + entry.getCount());
        }
        return String.join(", ", entries);
    }
}

File: manual/code/test_suites/cpp/FunctionBuildTest/src/function_collections_counter.cpp
#include <string>
#include <vector>
#include <map>
#include <sstream>
#include <regex>

class CounterCalculator {
public:
    std::string count_elements(const std::string& data) {
        if (data.empty()) return "failed";

        // 使用正则表达式分割字符串并过滤空元素
        std::regex regex("\\s+");
        std::vector<std::string> words(std::sregex_token_iterator(data.begin(), data.end(), regex, -1), std::sregex_token_iterator());

        if (words.empty()) return "failed";

        // 使用map保持元素插入顺序
        std::map<std::string, int> counter;
        for (const auto& word : words) {
            counter[word]++;
        }

        // 构建结果字符串
        std::vector<std::string> entries;
        for (const auto& entry : counter) {
            entries.push_back(entry.first + ":" + std::to_string(entry.second));
        }

        std::ostringstream result;
        for (size_t i = 0; i < entries.size(); ++i) {
            if (i != 0) result << ", ";
            result << entries[i];
        }

        return result.str();
    }
};